src/Illuminate/Http/Resources/Json/Resource.php

```diff
@@ -91,12 +91,12 @@ public function resolve($request = null)
             $request = $request ?: Container::getInstance()->make('request')
         );
 
-        if (is_array($data)) {
-            $data = $data;
-        } elseif ($data instanceof Arrayable || $data instanceof Collection) {
-            $data = $data->toArray();
-        } elseif ($data instanceof JsonSerializable) {
-            $data = $data->jsonSerialize();
+        if ($data instanceof Arrayable || $data instanceof Collection) {
+            return $this->filter((array) $data->toArray());
```

Comment:
Why are you casting it to an array again?

Reply:
Good spot. I only re-arranged the code, I didn't look too much into what it was doing. I will amend this